### ISSUE TITLE
Tracking issue for fix TODO comments

### ISSUE THREAD

**Issue by jiacai2050** [apache/incubator-horaedb#1466]
### Describe This Problem

Hi community, 

Our codebase contains numerous 200+ TODOs now! This indicates significant technical debt. Ignoring this will inevitably cause severe code rot.

As a community, we must acknowledge this problem affects our collective interests. While specific individuals may own portions of the code, we share responsibility for the overall health of our ecosystem.

The time to act is now. Let's work together to strengthen the foundation of our codebase before decay spreads further. Our future depends on it.

### Proposal

I will begin this task by review, list the most easy TODOs first, anyone interested are welcome to help us fixing them.

### Additional Context

1. [x] https://github.com/apache/incubator-horaedb/blob/63de16701c9f8614b1c7a52b15b9d2b8cbc3be59/src/benchmarks/src/merge_sst_bench.rs#L170
2. [x] https://github.com/apache/incubator-horaedb/blob/63de16701c9f8614b1c7a52b15b9d2b8cbc3be59/src/components/future_ext/src/retry.rs#L24
3. [x] https://github.com/apache/incubator-horaedb/blob/63de16701c9f8614b1c7a52b15b9d2b8cbc3be59/src/components/tracing_util/src/logging.rs#L129
4. [x] https://github.com/apache/incubator-horaedb/blob/63de16701c9f8614b1c7a52b15b9d2b8cbc3be59/src/table_engine/src/partition/rule/key.rs#L240
5. [ ] https://github.com/apache/incubator-horaedb/blob/63de16701c9f8614b1c7a52b15b9d2b8cbc3be59/src/components/parquet_ext/src/prune/min_max.rs#L91
6. [ ] https://github.com/apache/incubator-horaedb/blob/63de16701c9f8614b1c7a52b15b9d2b8cbc3be59/src/analytic_engine/src/instance/wal_replayer.rs#L377
7. [ ] https://github.com/apache/incubator-horaedb/blob/63de16701c9f8614b1c7a52b15b9d2b8cbc3be59/src/analytic_engine/src/manifest/details.rs#L379

**Comment by jiacai2050** [comment]
To avoid conflicts with others, please lease leave an comment below to tell us which TODO are you going to fix.

**Comment by Lethannn** [comment]
Working on "TODO: Replay logs of different tables in parallel. "

**Comment by zealchen** [comment]
Working on "TODO: make level configurable "
Working on "TODO: add backoff "

**Comment by jiacai2050** [comment]
@Lethannn  It's has been quiet for a while, I wonder if there are any problems when you implement this task?

**Comment by Lethannn** [comment]
> @Lethannn It's has been quiet for a while, I wonder if there are any problems when you implement this task?

Apologies for the silence lately. I've been dealing with some health issues over the past few weeks :( . Planning to tackle the task this weekend

**Comment by jiacai2050** [comment]
> I've been dealing with some health issues over the past few weeks

Sorry to hear that, let me know if you need any help.

**Comment by zealchen** [comment]
Working on "TODO: props should be an argument "  and "use NonZeroUsize "

**Comment by jiacai2050** [comment]
Tasks are all finished, thanks everyone involved.

Will open an new issue to track more TODO.
